x-pack/plugins/enterprise_search/public/applications/app_search/components/meta_engine_creation/index.ts

```diff
@@ -0,0 +1,8 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+export { MetaEngineCreation } from './meta_engine_creation';
```

Comment:
[Just thinking out loud, not a change request] What are your thoughts on putting the `MetaEngineCreation` and `MetaEngineCreationLogic` files in `engine_creation` rather than it's own folder? I'm kinda 50/50 on that myself, mostly just raised it because it feels like we're starting to trend towards "meta engines are still considered engines but a different type". Feel free to totally disregard this stream of consciousness comment unless you really like this idea for whatever reason

Reply:
Okay so I actually went through like, four phases of this PR:
1) I made the whole thing using a shared `EngineCreation` and `EngineCreationLogic` file that keyed off a `type` prop (kinda like how we are atm re-using `EngineTable` for both types of engines on `EngineOverview`
2) I made two seperate `EngineCreation` and `MetaEngineCreation` components, with a shared `EngineCreationLogic` file, just the logic file was keyed off a `type` prop (since the form was so simple, using a shared component with a `engineCreationTitle` `engineCreationDescription`, and passing through all the actions, as I was doing in 1, didn't feel readable to me)
3) Two seperate `EngineCreation` and `MetaEngineCreation`  components, with two seperate `EngineCreation` and MetaEngineCreation`, utilizing a shared `BaseEngineCreation` kea component (I don't like that two views were using a single file and each using 2/3 of it, with only 1/3 shared between the to.  There should be a way to seperate these concerns up.  But at this point I knew I was in over my head, and I had lost time because of snow)
4) Two seperate `EngineCreation` and `MetaEngineCreation`  components, with two seperate `EngineCreation` and MetaEngineCreation` logic files, they don't share anything.

The reason I went with number 4 ultimately is that I think its a cleaner migration, its easier to reason about for the purposes of migration (we don't have to debate whether I abstracted at the "right" level, which I think I've missed the mark on recently wrt stuff like `GenericEndpointInlineEditableTable`).

I _would_ like to combine these into some sort of shared component logic file, but I don't think the solution is "put a bunch of if statements everywhere" which is what i was trending towards.  I think the solution is somewhere between #2 and #3 honestly.  I just lost too much time trying to find it with results I didn't end up feeling confident in, so I just went with what I knew would be the most straightforward to review.

Okay that doesn't really directly answer your question:
> What are your thoughts on putting the MetaEngineCreation and MetaEngineCreationLogic files in engine_creation rather than it's own folder? 

Engine Creation and Meta Engine Creation are both top-level app views, and for the most part we seem to be giving each top-level view component its own directory under `components/`, especially when they don't share any code.  I think I'd like to merge all of this into one Engine Creation To Rule Them All but I think that time will be when we decide to a redesign of this page.  Moving this under `engine_creation` as a one off seems off to me, when they don't share anything beyond a visual and functional similarity.

Reply:
Thanks for the super detailed answer!! FWIW I'm a **super** strong +1 with the direction you went in (separate logic files rather than a ton of branches). I totally get the subsequent logic of keeping them in separate folders as well - I think my only very weak argument around combining them anyway was more the 'conceptual' side of things, but I think the view argument is a very strong one. Consider me sold! 👍